src/components/TestPage.jsx

```diff
@@ -1,20 +1,82 @@
-import React from 'react'
-
+import React, { Component } from 'react'
+import config from 'config/configFromCookie'
+import { maxHeaderSize } from 'http';
+const { apiURL } = config
 /*
  * TestPage: React frontend for --test functionality.
  *
  *
  *
 */
-function TestPage () {
-  return (
-    <div className='app-page'>
-      <h2>Test a Mapping</h2>
-      <p>
-        Please enter a valid DBQ number below, then choose the mapping file you want to test.
-      </p>
-    </div>
-  )
+class TestPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dbqnum: '',
+      uploadInput: null,
+      results: '',
+      isTesting: false,
+    };
+
+    this.handleUpload = this.handleUpload.bind(this);
+    this.baseState = this.state
+  }
+  resetForm = () => {
+    this.myFormRef.reset();
+    this.setState(this.baseState)
+  }
+
+  handleUpload(ev) {
+    ev.preventDefault();
+    this.setState({ isTesting: true })
+    const data = new FormData();
+    data.append('file', this.uploadInput.files[0]);
+    data.append('dbqnum', this.dbqnum.value);
+    fetch(apiURL+'test_dbq', {
+      method: 'POST',
+      body: data,
+    }).then(res => res.text())
+      .then(text => this.setState({ results: text, isTesting: false}), (error) => { 
+        if (error){
```

Comment:
Just a note, no action required, but if you don't have the API running, The error that you get "Cannot POST" doesn't get caught by this conditional